src/components/Image/index.tsx

```diff
@@ -33,17 +31,14 @@ function Image(props) {
     useEffect(() => {
         // If an onLoad callback was specified then manually call it and pass
         // the natural image dimensions to match the native API
-        if (onLoad == null) {
+        if (typeof onLoad !== 'function' || typeof source === 'number' || !source.uri) {
             return;
         }
         RNImage.getSize(source.uri, (width, height) => {
             onLoad({nativeEvent: {width, height}});
         });
     }, [onLoad, source]);
 
-    // Omit the props which the underlying RNImage won't use
-    const forwardedProps = _.omit(props, ['source', 'onLoad', 'session', 'isAuthTokenRequired']);
```

Comment:
Why did you remove this?

Reply:
I can achieve the same thing with `{source: propsSource, isAuthTokenRequired, onLoad, session, ...forwardedProps}: ImageProps`, so I thought it makes sense to do it all at once